### PR TITLE
FlexIO-driven LEDs

### PR DESCRIPTION
**Pull request by ggVGc**
This drives the LEDs nicely through FlexIO, fed by DMA in the main firmware.
- Works without flicker for me.
- Tested on `2.1` and `2.0-beta1`.
- Tested with clock over USB-MIDI, and shows no flicker.

Should close https://github.com/datomusic/duo-imxrt/issues/49 and https://github.com/datomusic/duo-imxrt/issues/138

### CODE REVIEW

**Comment by DavidMenting** [comment]
Alright, nice work! This is a really big improvement going forward.

The led driving part based on Finomnis/ws2812-flexio needs proper credit with a link to the original source code which I'll add.

The dma channel handling seems hardcoded for DMA0. Let's make that configurable. We might even be able to take the interface from Teensy's DMAchannel.h and put it to good use.

File: platform/brains2/cmake/lib.cmake
Comment (DavidMenting) [inline platform/brains2/cmake/lib.cmake:1]: https://github.com/datomusic/duo-imxrt/pull/144#:~:text=not%20in%20lib-,85795d2,-Merge%20state

**Comment by ggVGc** [comment]
@DavidMenting  yeah, I think this should get at least one refactoring pass before merging to `main`, making for example the DMA channel configurable.
But at least this is a known state of things working.

**Comment by DavidMenting** [comment]
I don't know if the refactoring of DMA channel is necessary before merging back into main as it should not make any difference right now. We could add it as a separate issue and make the I2S audio DMA use the same interface.

**Pull request merged**